packages/store/src/BlobStorage.types.ts

```diff
@@ -0,0 +1,35 @@
+export interface BlobRef {
+  blobId: string;
+}
+
+export type BlobInput = {
+  data:
+    | {
+        textContent: string;
+        encoding: 'utf8';
+      }
+    | Blob;
+  // TODO:
+  // | BufferSource
+  // | ReadableStream<any>;
+  mimeType?: string;
+};
+
+export interface BlobStorage {
+  /** Upload a new blob */
+  upload(blobInput: BlobInput): Promise<BlobRef>;
+  /** Get info about this blob, some may be specific to the storage provider. */
+  getDebugInfo(ref: BlobRef): Promise<Record<string, unknown>>;
+  /** Get a web accessible URL to access this BlobRef's contents, such as for an `<img src={blobURL}/>` */
+  getWebURL(ref: BlobRef): Promise<URL>;
+  // // Hmm: How will we handle exporting large files / lots of files?
+  // // We shouldn't just glob together everything into a big JSON, right?
+  // // Should we require exporting by individual resource ids?
+  // // Should exporting be a part of a higher interface on a per storage basis?
+  // // e.g. export for a Torrent or IPFS stored resources might actually export Content Hashes
+  // // and import takes those to re-associate them...
+  // // But, that feels like a specific for the specific storage.
+  // /** Export to something... need goals defined */
+  // export(): Promise<Record<string, unknown>>;
+  // import(): Promise<void>;
```

Comment:
I'm not entirely sure what the context is of the `export` + `import`. Is this supposed to look similar to anything that already exists?

Reply:
`export` means to dump the storage content to user file system (for backup and data ownership), just like what the `export workspace` feature in https://livedemo.affine.pro/ is doing. And with `import` we load them back.